migrate.js

```diff
@@ -14,60 +27,142 @@ const octokitTarget = new Octokit({
   auth: process.env.GH_PAT_TARGET,
 });
 
-const secretValue = "temp"
-let newEnvs = []
+const secretValue = "temp" // Set the secret value to encrypt the secrets, temp is used since secrets cannot be migrated using this script
+let newEnvs = [] // Needed to store the encrypted secrets
 
+console.log("Migrating environments from source to target repo...");
+// Console log exist all over the program, these can be uncommented to see the output of the program
+
+// Function to migrate environments
 async function migrateEnvironments() {
-  const {data: environments} = await octokitSource.repos.getAllEnvironments(sourceRepo)
-  console.log("Environments: ", environments)
-
-  // let usersMap = new Map()
-  // try {
-  //   usersMap = await generateUserMap()
-  //   console.log("generate user map: ")
-  //   console.log(usersMap)
-  // } catch(error) {
-  //   console.error(error)
-  // }
+  const {data: environments} = await octokitSource.repos.getAllEnvironments(sourceRepo);
+  //console.log("Environments: ", environments);
+
+  // Create a copy of the environments object environments list
+  let tempEnv = environments.environments;
+
+  // Grab the repo ID for the target repo
+  const repoId = await octokitTarget.rest.repos.get({
+    owner: targetRepo.owner,
+    repo: targetRepo.repo,
+  });
 
+  // Update environments to include REPO ID as a property
+  tempEnv = tempEnv.map(environment => {
+    return {
+      ...environment,
+      repoID: repoId.data.id // replace 'yourRepoID' with the actual repoID
+    };
+  });
+
+  //console.log("environments NEW: ", tempEnv);  
+
+  // Create new users map
+  let usersMap = new Map()
+  try {
+    usersMap = await generateUserMap()
+    //console.log("user map: ")
+    //console.log(usersMap)
+  } catch(error) {
+    console.error(error)
+  }
+
+  // Create empty reviewer list and environment list for use in later functions
   let reviewerList = []
   let envList = []
 
-  for (const env of environments.environments) {
+  // itterate through the environments and create the environments in the target repo
+  for (const env of tempEnv) {
+    //console.log("Environment: ", env)
```

Comment:
We can probably log the environments so they show in the logs

Reply:
done